### ISSUE TITLE
Tech Reborn: Performance Issues

### ISSUE THREAD

**Issue by mtimyself** [FTBTeam/FTB-Continuum#140]
* **Modpack Version**: 1.0.1
* **Issue**:

There appears to be a major performance problem involving Tech Reborn machinery.
It can most easily be replicated and observed with the Electric Furnace but other overclockable machinery exhibits the same issue, just less dramaticly.
A fully upgraded (4 Overclockers) Electric Furnace will absolutely tank the client's frames (sub 10 on a decent rig) if using the machine's own import and export functionality.
If the machines starts with an empty inventory and you dump a couple stacks into the input chest it will not be able to keep itself supplied properly since the processing speed exceeds the input speed.
This shoots up chunk updates to constantly over 40 and up to 90 and renders the client nearly irresponsive.

Example:
http://prntscr.com/jqm5xh

Other machinery also exhibits similar problems, yet it seems to not be related to the import and export feature.
Here is the Grinder with 4 Overclockers:
http://prntscr.com/jqmc39

It is interesting to note that if the Furnace is provided with other means of input (e.g. piping it in or manually placing something inside to be processed) it does not seem to have any negative (noticeable) impact on performance (even if automatic output is enabled) while the Grinder will continue to exhibit 
these very regular spikes when working no matter if an automatic input and/or output is configured.
Other machines (tested Alloy Smelter, Compressor etc.) also behave like the Grinder.

E: After some testing it would appear that the issue is localized to only our base on the server but highly replicable there.
Nothing has been changed about or added to the pack.

**Comment by modmuss50** [comment]
I’ll take a look when I get a chance. My guess is the machines are being turned on and off very quickly even when it can keep processing.

Make sure you have forge’s threaded chunk rendering enabled as that may help.

**Comment by mtimyself** [comment]
I can confirm this is happening. In the Debug overlay you can see the machine's state flickering from active to inactive very rapidly.


E: This alone is not it though, I just tested it in SP Creative and there was no impact at all.
There must be an interaction with something else at play here.
I will tp to different coordinates on the server and give it a go there.

E2: I am having problems recreating the issue outside our base but as you can see from the above screenshots it is a major issue we are facing right now.
There must be something at work that is not immediately apparent.

**Comment by tyler489** [comment]
i to am having this issue... was wondering why my frames dropped in our base..

**Comment by jottafan** [comment]
My base is unbearable when running tech reborn machinery.

**Comment by modmuss50** [comment]
Is the grinder or machine getting enough power? As in SP with a creative power gen I cannot reproduce this issue.

![java_2018-06-05_19-54-17](https://user-images.githubusercontent.com/4324090/40996750-8659c7c4-68fa-11e8-9a07-e24f0b323af9.png)
![java_2018-06-05_19-59-11](https://user-images.githubusercontent.com/4324090/40996931-0c6463ec-68fb-11e8-8606-63265bb1b6d2.png)

I then created a test with 2 machines going into each other:

![java_2018-06-05_19-58-53](https://user-images.githubusercontent.com/4324090/40996938-0f17cb60-68fb-11e8-9c81-56a480b2d9c6.png)

![java_2018-06-05_19-55-14](https://user-images.githubusercontent.com/4324090/40996754-8961cce6-68fa-11e8-9315-fb7b3f21f0c6.png)

If the machine isn't getting enough input, or the output isn't being extracted fast enough then the machine will turn on and off. Can you try and make a smaller easier for me to reproduce setup?

Thanks in advance.

**Comment by tyler489** [comment]
I have xnet pushing power in and have an ae buss pulling items out. Have xnet pushing stacks of items in from 2 tier 6 ore miners.

**Comment by mtimyself** [comment]
The Electric Furnace is especially broken for us.
This one entirely kills your client but only when using automatic input and output. If the material is provided by other means it is the only TR machine that works properly.

All other machines I have tested (e.g. Grinder) has a problem no matter what and exhibits extremely regular spikes.
My setup is fed by "oldschool" resonant fluxducts still but afaik a Low Tier machine should not be able to use anything near that, even with 4 Overclockers. The energy bar remains filled entirely.

I can only reproduce the thing inside the base (or even just workshop). That said tps are stable and it is not THE "busiest" chunk in base.

**Comment by modmuss50** [comment]
With the help of @mtimyself I have tweaked the machines to input items faster, this prevents them from running out of items causing the machine to turn off. The electric furnace is also ever so slightly slower with 4 upgrades in now.

@tfox83 I suggest changing forge's threaded chunk rendering to be enabled by default as it helps a lot in areas where there are many block updates.

DW20 did a video on it showing the benefits to it and how to enabled it, https://www.youtube.com/watch?v=t-BgOlJ8N4U

**tldr;**

New updates for TR and RC make it better, and enabled threaded chunk rendering makes things a lot better overall
